sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -347,7 +347,9 @@ def _handle_exception(self, exception: BaseException) -> "ServiceBusError":
             # failure modes, that motivates us to disallow this.
             if self._session and self._running and isinstance(error, (SessionLockLostError, ServiceBusConnectionError)):  # type: ignore
                 self._session._lock_lost = True  # type: ignore
-                self._close_handler()
+                if self._amqp_transport.KIND == "uamqp":
+                    # Only close for uamqp, we let pyamqp close the connection via the retry loop.
+                    self._close_handler()
```

Comment:
Do we need to close the handler for pyamqp anymore because we get the error now from the service (aka we get detach flow)

Reply:
will moving this into transport change the flow of when this gets called ? In general yes, uAMQP specific stuff like this should head there